celltype_annotation/analysis/SingleR-combining-refs.Rmd

```diff
@@ -0,0 +1,203 @@
+  dplyr::mutate(top_labels = forcats::fct_lump_n(pruned.labels, 10)) |>
+  dplyr::count(top_labels, reference)
+
+# same barchart but with just the top 10 labels
+ggplot(top_label_counts_df, aes(x = reference, fill = top_labels, y = n)) +
+  geom_bar(position = "stack", stat = "identity") +
+  labs(x = "",
+       y = "Number of cells",
+       label = "Cell type") +
+  guides(x = guide_axis(angle = 90))
+```
+
+It looks like most of the labels are coming from monocyte progenitors in BlueprintEncodeData reference with a few cells being assigned to other labels present in the other datasets.
+
+Below, I looked at how the labels looked when using all references vs. single references. 
+When running SingleR with multiple references, first DE genes are identified within each reference and labels assigned within each reference and then statistics are merged across references (I'm still slightly unclear on exactly what that means).
+
+```{r, fig.height=5}
+# get all the labels from combined and each individual reference
+all_results <- result_df |> 
+  dplyr::select(ends_with("pruned.labels")) |>
+  tidyr::pivot_longer(cols = everything(), 
+                      values_to = "label.ont",
+                      names_to = "reference") |> 
+  # rename final assigned labels as combined and remove some extra strings
+  dplyr::mutate(reference = ifelse(reference == "pruned.labels", "combined", reference),
+                reference = stringr::str_remove_all(reference, "orig.results.|.pruned.labels")) |>
+  # grab top 10 and relevel for ordering |>
+  dplyr::mutate(top_labels = forcats::fct_lump_n(label.ont, 10), 
+                reference = factor(reference, 
+                                   levels = c("combined", 
+                                              "BlueprintEncodeData", 
+                                              "HumanPrimaryCellAtlasData",
+                                              "DatabaseImmuneCellExpressionData",
+                                              "MonacoImmuneData"))) |>
+  dplyr::count(top_labels, reference)
+
+# create a barchart to look at assignments between combined vs. single refs 
+ggplot(all_results, aes(x = reference, fill = top_labels, y = n)) +
+  geom_bar(position = "stack", stat = "identity")  +
+  labs(x = "",
+       y = "Number of cells",
+       label = "Cell type") +
+  guides(x = guide_axis(angle = 90))
+  
+  
+```
+
+It looks like the combined references look the most similar to the BlueprintEncodeData. 
+However, if we used individual references we see that the other references assign cells as CD34+ progenitors or monocytes. 
+In the combined reference we also see a small portion of the cells are annotated as CD34+ progenitors, which would not have been the case if we only used BlueprintEncodeData.
+
+```{r fig.height=20, fig.width=15}
+# make some heatmaps because why not
+SingleR::plotScoreHeatmap(combined_result,
+                          grid.vars = list(ncol = 1))
```

Comment:
It is annoying that these heatmaps are all different sizes, which makes comparison among them difficult. I think we may be able to address this by using a set of labels found in all references, if they exist, and the `labels.use` parameter.

Some other parameters we might try from `pheatmap`, if you haven't already:

`annotation_legend = FALSE` to remove the legend.
We might also want to get rid of the row cluster tree with `treeheight_row = 0`